### DISCUSSION TITLE
How to handle PrismaClientRustPanicError

### DISCUSSION THREAD

**Discussion by yaacov-commenda** [prisma/prisma#18697] [Q&A] [answered]
The docs say 

> Prisma Client throws a PrismaClientRustPanicError exception if the underlying engine crashes and exits with a non-zero exit code. In this case, the Prisma Client or the whole Node process must be restarted.

If I catch a `PrismaClientRustPanicError`, should I restart the prisma client or the Node process? If so, how do I know which one to restart? If it's the Prisma client that needs to be restarted, how do I do that?

**Comment by ludralph** [comment, marked as answer]
Hi @yaacov-commenda 👋 

If you catch a PrismaClientRustPanicError, it means that the underlying Prisma engine has crashed and exited with a non-zero exit code. In this case, you should restart the whole Node process, as the crash may have affected other parts of your application as well.

To restart the Node process, you can use a process manager like PM2 or Forever. These tools can automatically restart your Node application if it crashes or if the process is killed.

If you're not using a process manager, you can manually restart the Node process by stopping and starting it again.
